src/main/java/wooteco/subway/domain/Sections.java

```diff
@@ -0,0 +1,174 @@
+package wooteco.subway.domain;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class Sections {
+
+    private final LinkedList<Section> sections;
+
+    public Sections(LinkedList<Section> sections) {
+        this.sections = sections;
+    }
+
+    public Sections(List<Section> sections) {
+        this.sections = new LinkedList<>(sections);
+    }
+
+    public Sections(Section section) {
+        this.sections = new LinkedList<>();
+        sections.add(section);
+    }
+
+    public static Sections from(List<Section> mixedSections) {
+        LinkedList<Section> sections = new LinkedList<>(List.of(mixedSections.remove(0)));
+        while (!mixedSections.isEmpty()) {
+            for (int i = 0; i < mixedSections.size(); i++) {
+                Section section = mixedSections.get(i);
+                if (sections.getFirst().canUpExtendBy(section)) {
+                    sections.addFirst(mixedSections.remove(i));
+                    break;
+                }
+                if (sections.getLast().canDownExtendBy(section)) {
+                    sections.addLast(mixedSections.remove(i));
+                    break;
+                }
+            }
+        }
+        return new Sections(sections);
+    }
+
+    public Section findTop() {
+        return sections.getFirst();
+    }
+
+    public Section findBottom() {
+        return sections.getLast();
+    }
+
+    public Station findTopStation() {
+        return findTop().getUp();
+    }
+
+    public Station findBottomStation() {
+        return findBottom().getDown();
+    }
+
+    public void add(Section section) {
+        if (section.isAlreadyIn(getStations())) {
+            throw new IllegalArgumentException("이미 포함 된 구간입니다.");
+        }
+        if (canExtendBy(section)) {
+            extendSections(section);
+            return;
+        }
+        if (canAnyDivideBy(section)) {
+            addSectionInside(section);
+            return;
+        }
+        throw new IllegalArgumentException("겹치는 역이 없이 추가할 수 없습니다.");
+    }
+
+    private void addSectionInside(Section section) {
+        for (int i = 0; i < sections.size(); i++) {
+            Section origin = sections.get(i);
+            if (origin.isSameUpStation(section) || origin.isSameDownStation(section)) {
+                sections.remove(i);
+                sections.addAll(i, origin.divideBy(section));
+                return;
+            }
+        }
```

Comment:
```suggestion
     method1
        for (int i = 0; i < sections.size(); i++) {
            Section origin = sections.get(i);
             if (origin.isSameUpStation(section) || origin.isSameDownStation(section)) {
                 return i;
             }
        }
       
       method2
        sections.remove(i);
        sections.addAll(i, origin.divideBy(section));
```
저는 이렇게 2 부분으로 나누는 방법을 생각했습니다!
어떻게 하는게 나을지는 써머가 생각해보고 정하면 될 것 같아요ㅎㅎ